test-suite/js/src/test/scala/org/scalajs/testsuite/jsinterop/MapTest.scala

```diff
@@ -0,0 +1,104 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.testsuite.jsinterop
+
+import org.junit.Assert._
+import org.junit.Assume._
+import org.junit.{BeforeClass, Test}
+import org.scalajs.testsuite.utils.AssertThrows._
+import org.scalajs.testsuite.utils.Platform
+
+import scala.scalajs.js
+
+object MapTest {
+  @BeforeClass
+  def assumeRuntimeSupportsMap(): Unit = {
+    assumeTrue("Assume Map exists in Global",
+        js.typeOf(js.constructorOf[js.Map[_, _]]) != "undefined")
+  }
+}
+
+class MapTest {
+
+  // scala.scalajs.js.Map
+
+  @Test def testApply(): Unit = {
+    val obj = js.Map("foo" -> "bar")
+    assertThrows(classOf[NoSuchElementException], obj("bar"))
+    assertEquals("bar", obj("foo"))
+    assertEquals(1, obj.size)
+
+    val empty = js.Map()
+    assertEquals(0, empty.size)
+  }
+
+  @Test def testGet(): Unit = {
+    val obj = js.Map.empty[String, Int]
+    obj("hello") = 1
+
+    assertTrue(obj.get("hello").get == 1)
+    assertFalse(obj.get("world").isDefined)
+  }
+
+  @Test def testMinsuEqual(): Unit = {
+    val obj = js.Map("a" -> "A")
+    assert(!obj.delete("b"))
+    assertEquals(1, obj.size)
+
+    assert(obj.delete("a"))
+    assertEquals(0, obj.size)
+  }
+
+  @Test def testIterator(): Unit = {
+    val obj = js.Map("foo" -> 5, "bar" -> 42, "babar" -> 0)
+    var elems: List[(String, Int)] = Nil
+    for ((prop, value) <- obj) {
+      elems ::= (prop, value)
+    }
+    assertEquals(3, elems.size)
```

Comment:
This is implied by the test with `sameElements`, so there's no need for it either.